fecgov/fec-cms
# Add category to doc page drpdwn/fix ResourcePg filtering

## Summary

- Resolves #4605

- Add `Other reports and plans` category to `DocumentPage category dropdown` and  `ResourcePage > Promote tab > category dropdown`
- Fix problem found where ResourcePages Report (created as a child of a `Reports about FEC section` and given  a category in the Promote tab) shows up in all categories/years when a user filters reports. See screenshots 2 and 3.
     - **History:**  Support for adding resource pages as reports was added in this commit, but filtering was overlooked: 
PR: https://github.com/fecgov/fec-cms/pull/888
Commit: https://github.com/fecgov/fec-cms/commit/efb57d7ce62d4597c64599e5c279af93bdd914bc#diff-3266eb159536d74bff5241aee52e813968074331615b951d5acab2470253d526

### Required reviewers

One  front end reviewer

## Impacted areas of the application

**modified:**  fec/constants.py
**modified:**  home/templatetags/document_feed.py
**new file:**    home/migrations/0115_auto_20210506_2326.py



## Screenshots
<hr>

### 1) Add new category:
<hr>
<img width="885" alt="Screen Shot 2021-05-06 at 11 58 33 PM" src="https://user-images.githubusercontent.com/5572856/117396086-3ec8bd00-aec7-11eb-929e-5a7a50d8a432.png">

<hr>


### 2) Before: ResourcePage reports are not filtered:
<hr>
<img width="624" alt="Screen Shot 2021-05-06 at 11 55 33 PM" src="https://user-images.githubusercontent.com/5572856/117396245-8cddc080-aec7-11eb-88c0-f54881be11f7.png">

<hr>

### 3) After:ResourcePage reports are filtered
<hr>
<img width="877" alt="Screen Shot 2021-05-06 at 11 59 25 PM" src="https://user-images.githubusercontent.com/5572856/117396016-20fb5800-aec7-11eb-9e96-a37ee5de0d3e.png">

<hr>

## How to test

- Checkout `feature/4605-other-reports-category`
- `./manage,py migrate`
- Create a DocumentPage or edit an existing page as child of `Agency Operations` and choose `Other reports and plans` category
- Confirm that the category works on the Reports about the FEC > Agency operations page
- Confirm that the `ResourcePage , 2016 E-Filing study` , only shows up when filtering by `Study` or `All`
- Confirm `year` filtering works on the `ResourcePage , 2016 E-Filing study` as well. Note: Its pub date is 2017.
